src/ast.rs

```diff
@@ -1,18 +1,79 @@
+use std::fmt::Display;
+
 use crate::{ByteString, TokenKind};
 
 pub type Block = Vec<Statement>;
 pub type Program = Block;
 
+#[derive(Debug, Eq, PartialEq, Clone)]
+pub enum TryBlockCaughtType {
+    Identifier(ByteString),
+    Union(Vec<ByteString>),
+}
+
 #[derive(Debug, Eq, PartialEq, Clone)]
 pub enum Type {
-    Plain(ByteString),
+    Identifier(ByteString),
     Nullable(ByteString),
```

Comment:
TODO: nullable shouldn't exist, should be represented as `Union(Null, T)` instead.

Reply:
I'm not sure on this. I've mostly been modelling AST structure off of Nikita's PHP-Parser, which has a specific `NullableType` object.

Reply:
if the plan is to be able to re-construct the source code, yes, otherwise, it doesn't make sense, as with `iterable -> array|Traversable`.

Reply:
Yeah, I want the AST to represent the code as it was written ideally.